### DISCUSSION TITLE
[PSD12] Exercise 3.10

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/psd2223#177] [Esercizi (A-L)]
**Exercise 3.10** A _JK flip-flop_ receives a clock and two inputs, $J$ and $K$. On the rising edge of the clock, it updates the output, $Q$. If $J$ and $K$ are both 0, $Q$ retains its old value. If only $J$ is 1, $Q$ becomes 1. If only $K$ is 1, $Q$ becomes 0. If both $J$ and $K$ are 1, $Q$ becomes the opposite of its present state.

(a) Construct a JK flip-flop using a D flip-flop and some combinational logic.
(b) Construct a D flip-flop using a JK flip-flop and some combinational logic.
(c) Construct a T flip-flop (see **Exercise 3.9**, #176 ) using a JK flip-flop.

**Comment by Elia-Belli** [comment]
- **JK flip-flop from a D flip-flop**
![D](https://user-images.githubusercontent.com/107944476/211644767-bd454c74-1291-4e9c-9183-55e641f4c9da.jpg)

- **D flip-flop from a JK flip-flop**
![D from JK](https://user-images.githubusercontent.com/107944476/211643431-c8eeb8c7-951b-4211-886a-fe217f22a2d0.jpg)

- **T flip-flop from a JK flip-flop**
![T](https://user-images.githubusercontent.com/107944476/211645008-fb24baab-c8de-4ac9-877a-42d87c540914.jpg)

**Comment by fraxb04** [comment]
corretto
